crypto/data_formats.go

```diff
@@ -15,41 +15,72 @@
 package crypto
 
 import (
-	"encoding/base64"
 	"fmt"
-	"strconv"
 
-	"github.com/benlaurie/objecthash/go/objecthash"
-	"github.com/google/trillian"
+	"github.com/google/certificate-transparency-go/tls"
+
+	pb "github.com/google/trillian"
 )
 
 // This file contains struct specific mappings and data structures.
-// TODO(gdbelvin): remove data-structure specific operations.
 
-// Constants used as map keys when building input for ObjectHash. They must not be changed
-// as this will change the output of hashRoot()
+type logVersion int
+
+const (
+	// LogRootV0 indicates version 0 for the canonical representation of the SignedLogRoot.
+	LogRootV0 logVersion = iota
+)
+
+// CanonicalLogRoot returns a canonical TLS serialization of the log root.
+func CanonicalLogRoot(r *pb.SignedLogRoot, version logVersion) ([]byte, error) {
+	switch version {
+	case LogRootV0:
+		root := struct {
+			RootHash       []byte `tls:"minlen:0,maxlen:128"`
+			TimestampNanos uint64
+			TreeSize       uint64
+			LogID          uint64
```

Comment:
I think it's worth thinking about what a non-Go client of Trillian would need in order to validate signatures.  The relevant information is here in this struct; maybe it should be an exported `type` for more visible documentation and to make it more obvious when changes occur to it.   Also/alternatively, you could include the TLS schema in RFC 5246 style as a comment.

Reply:
I might need some help with the TLS schema.